docs/plugins.md

```diff
@@ -23,43 +27,36 @@ const myAdaptor = {
       }
     }
   }
 };
 ```
 
-Adaptors are simply the translation and sync layers between your custom data source and Ractive instances. The basic principle of an adaptor is as follows:
-
-1. Provides an POJO version of your data source to Ractive.
-2. Captures data changes on your data source and mirror them to the data in Ractive.
-3. Captures data changes on the data in Ractive and mirror them to the data source.
-
-Whether it's a third-party data modelling library, a RESTful service, a socket server, browser storage, or whatever, as long as all of the three can be done, it can be adapted.
-
-`filter` is a function that gets called to check if `object` needs to use an adaptor.
-
-`object` is the data source to adapt.
-
-`keypath` is the keypath to `object`.
-
-`ractive` is the ractive instance that is currently using the adaptor.
-
-`wrap` is a function that gets called to set up the adaptor on `object`.
-
-`prefixer` is a helper function that accepts an object and automatically prefixes `keypath` to the object's keys.
-
-`get` is a function that gets called when Ractive needs the adapted representation of the `object`.
-
-`set` is a function that is called when `ractive.set()` updates a keypath to a property of the adapted data. This function allows you to update the same property on `object`.
-
-`property` is the keypath to the property being updated, relative to `keypath`.
+An adaptor definition is an object with two methods:
 
-`value` is the value being passed into `ractive.set()`.
+- `filter`: A function which, if it returns `true`, tells Ractive to use the adaptor on the value.
+    - `value`: The value to check.
+    - `keypath`: The keypath of `value` relative to the instance root.
+    - `ractive`: The instance where `value` resides.
+- `wrap`: A function that sets up an adaptor.
+    - `ractive`: The instance where `value` resides.
+    - `value`: The value to adapt.
+    - `keypath`: The keypath of `value` relative to the instance root.
+    - `prefixer`: A function that accepts an object and returns a shallow clone with its keys prepended with `keypath`.
 
-`reset` is a function that is called when `ractive.set()` updates a keypath to the adapted data. This function allows you to either update `object` or tear down the adaptor.
+The `wrap` function must return an object with four methods:
 
-`teardown` is a function called when the adaptor is being removed. This function allows you to do cleanup work on anything that was done during the adaptor setup.
+- `get`: A function that returns the POJO representation of the adapted data.
+- `set`: A function that's called when the keypath being updated is a property of the adapted data.
+    - `property`: The property name to be updated.
+    - `value`: The new value.
+- `reset`: A function that's called when the keypath being updated is the adapted data itself. If `reset` returns `false`, Ractive will replace the data on the keypath with the POJO version, and tear down the adaptor.
```

Comment:
Did not read the source all the way. Not returning `false`, Ractive will set the new value to the path and update the adapted value.